wavecrestweb/kenworthy-machine
# #KM-34/feature quote page

## Description
This PR adds a responsive 'request a quote' page with a submittable form. Note: This page's form intentionally does not include the additional icons and subheading seen in the Figma design as that allowed me to easily reuse the form component the home page.


## Related Tickets & Documents
Closes #KM-34


## Acceptance Criteria

- [x] All fields must validate inputs (email format for 'Email', required fields for 'Name' and 'Email'). Display errors if validation fails.
- [x] Show a confirmation message upon successful form submission.
- [x] The form and all its components adjust properly across devices, with the vertical layout activating on tablet and smaller screens.





## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
| ✓ | :sparkles: New feature     |
|  ✓ | 🎨 Style                   |
|     | :hammer: Refactoring       |
|     | 🔥 Performance Improvements|
|     | :bug: Bug fix              |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |


## Mobile & Desktop Screenshots/Recordings
#### Laptop: 
![Screenshot 2024-04-30 at 1 42 14 PM](https://github.com/wavecrestweb/kenworthy-machine/assets/115492619/6e698236-9401-4a21-bf90-8e94efdc6ed6)

#### Tablet: 
![Screenshot 2024-04-30 at 1 39 51 PM](https://github.com/wavecrestweb/kenworthy-machine/assets/115492619/9e558e22-b263-4f03-8570-59f5732cc3db)

#### Mobile:
![Screenshot 2024-04-30 at 1 41 00 PM](https://github.com/wavecrestweb/kenworthy-machine/assets/115492619/839211d8-2a38-4352-89e9-40978cda4ab0)




## What I learned
I practiced setting up the TinaCMS connection again, which operated a little differently when I was passing some the CMS content as props and some needed to be used directly in the component. 

## Testing steps
1. `git switch KM-34/feature-quote-page`
2. `npm i`
3. Create a `.env` file with the following variables:
```
EMAIL_TO= #test with any gmail account you have access to
EMAIL_USER= #test with any gmail account you have access to
EMAIL_PASSWORD= #this password needs to be specifically generated for use here by first turning on 2FA at [myaccount.google.com](https://myaccount.google.com/) and going the 'Security' section and then going to [myaccount.google.com/apppasswords](https://myaccount.google.com/apppasswords)
```
4. `npm run dev`
5. Go to localhost:3000/admin and click 'request a quote' in the header
6. Fill out the form with sample data and submit.
7. Verify that the form submits, and email is received, and that the successful submit message is shown.
8. Verify that the layout adjusts in smaller screen sizes.


## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
|     | 🙅 no documentation needed    |


## What gif best describes this PR?
  ![gif name](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExcWw3azdobXY4ZjFqdnVncDU4emJveDhvNHNudXozY3R2MXJlcW00byZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/2GN4OAIdsMDv10jAVn/giphy.gif)
